doc/guides/build/index.rst

```diff
@@ -0,0 +1,154 @@
+directory, which refers to the *CMakeLists.txt* file in the Zephyr
+top-level directory, which in turn refers to *CMakeLists.txt* files
+throughout the build tree (directly and indirectly). Its primary
+output is a set of Makefiles  to drive the build process, but *CMake*
+scripts do some build processing of their own:
+
+Device tree
+   Using *cpp*, device-tree specifications (*.dts/.dtsi* files) are
+   collected from the target’s architecture, SoC, board, and
+   application directories and compiled with *dtc*. Then the build
+   tool (scripts/dts) convert this into *.h* files for later
+   consumption.
+
+Device tree fixup
+   Files named *dts_fixup.h* from the target’s architecture, SoC,
+   board, and application directories are concatenated into a single
+   *dts_fixup.h*. Its purpose is to normalize constants output in the
+   previous step so they have the names expected by the source files
+   in the build phase.
+
+Kconfig
+   The build tool reads the *Kconfig* files for the target
+   architecture, the target SoC, the target board, the target
+   application, as well as *Kconfig* files associated with subsystems
+   throughout the source tree. It incorporates the device tree outputs
+   to allow configurations to make use of that data. It ensures the
+   desired configuration is consistent, outputs *autoconf.h* for the
+   build phase.
+
+Build Phase
+***********
+
+The build phase begins when the user invokes *make*. Its ultimate
+output is a complete Zephyr application in a format suitable for
+loading/flashing on the desired target board (*zephyr.elf*,
+*zephyr.hex*, etc.) The build phase can be broken down, conceptually,
+into four stages: the pre-build, first-pass binary, final binary, and
+post-processing.
+
+Pre-build occurs before any source files are compiled, because during
+this phase header files used by the source files are generated.
+
+Pre-build
+=========
+
+Offset generation
+   Access to high-level data structures and members is sometimes
+   required when the definitions of those structures is not
+   immediately accessible (e.g., assembly language). The generation of
+   *offsets.h* (by *gen_offset_header.py*) facilitates this.
+
+System call boilerplate
+   The *gen_syscall_header.py*, *parse_syscalls.py* and
+   *gen_syscall_header.py* scripts work together to bind potential
+   system call functions with their implementations.
+
+.. figure:: build-build-phase-1.png
+    :align: center
+    :alt: Zephyr's build stage I
+    :figclass: align-center
+    :width: 80%
+
+First-pass binary
+=================
+
+Compilation proper begins with the first-pass binary. Source files (C
+and assembly) are collected from various subsystems (which ones is
+decided during the configuration phase), and compiled into archives
+(with reference to header files in the tree, as well as those
+generated during the configuration phase and the pre-build stage). If
+memory protection is enabled, then:
```

Comment:
Nit: In the rendered doc, it is not directly clear what bellow is under the "if memory protection is enabled", as "partition grouping" just seems like a new subsection.
Maybe it would be clearer if this "if .."  was in the paragraph below (?)

Reply:
I can emphasize it.

Reply:
I certainly can do it. Thanks for spotting it.